eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/index.js

```diff
@@ -47,58 +59,127 @@ export class UnwrappedNavigationSidebar extends Component {
     canAddQuestionConfirmation: PropTypes.bool.isRequired,
     canAddCalculatedSummaryPage: PropTypes.bool.isRequired,
     canAddQuestionPage: PropTypes.bool.isRequired,
+  },
+};
+
+const initialState = { label: true, controlGroup: [] };
+
+export const actionTypes = {
+  updateGroup: "updateControlGroup",
+  toggleLabel: "toggleLabel",
+  toggleAll: "toggleAllAccordions",
+  setInitial: "setInitialControlGroup",
+};
+
+export const reducer = (state, action) => {
+  switch (action.type) {
+    case actionTypes.updateGroup: {
+      const { identity, isOpen } = action.payload;
+      const updateAccordionState = state.controlGroup
+        .filter(x => x.identity !== identity)
+        .concat({ identity, isOpen: !isOpen });
+
+      const allOpen = updateAccordionState.every(x => x.isOpen === false);
+      return { controlGroup: updateAccordionState, label: allOpen };
+    }
+    case actionTypes.toggleAll: {
+      const toggleAll = state.controlGroup.map(x => ({
+        identity: x.identity,
+        isOpen: state.label,
+      }));
+
+      return { label: !state.label, controlGroup: toggleAll };
+    }
+    case actionTypes.setInitial: {
+      return { label: true, controlGroup: action.payload };
+    }
+    default:
+      throw new Error(`${action.type} isn't a valid dispatch type`);
```

Comment:
We might need to escape the apostrophe here for when/if this error is ever shown. Think our linter throws a bit of a wobbly. Not sure if that's just locally though as in production it shouldn't matter.